### PR TITLE
Fix AclVoter parameter can be not an object

### PR DESCRIPTION
**Pull request by jenyak**
I am targeting this branch, because it is a bug fix.

Closes [#4518](https://github.com/sonata-project/SonataAdminBundle/issues/4518)

## Changelog
```Markdown
### Fixed
- AclVoter parameter can be not an object
```

### CODE REVIEW

**Comment by jenyak** [comment]
@greg0ire Should this be a single commit?

**Comment by greg0ire** [comment]
Please fix StyleCI

**Comment by greg0ire** [comment]
Oh wow you just reverted the Style fix? Why?

**Comment by jenyak** [comment]
Little familiar with git commits manipulations. Fix it

**Comment by greg0ire** [comment]
Oh but it says there are conflicts. You should probably rebase

**Comment by jenyak** [comment]
you mean back before StyleCI commint ?

**Comment by jenyak** [comment]
why StyleCI doing `strlen($baseRole) == 0` to `0 == strlen($baseRole)`. `strlen($baseRole)` is not something you can accidentally set.

**Comment by greg0ire** [comment]
Just so you know, I couldn't agree more : https://dev.to/greg0ire/why-using-yoda-conditions-you-should-probably-not You should rebase on the latest 3.x

**Comment by jenyak** [comment]
I did rebase

File: Security/Authorization/Voter/UserAclVoter.php [hunk 1]
```diff
@@ -31,21 +31,21 @@ public function supportsClass($class)
...
     {
-        if (!$this->supportsClass(get_class($object))) {
+        if (!is_object($subject) || !$this->supportsClass(get_class($subject))) {
             return self::ACCESS_ABSTAIN;
         }
 
         foreach ($attributes as $attribute) {
-            if ($this->supportsAttribute($attribute) && $object instanceof UserInterface && $token->getUser() instanceof UserInterface) {
-                if ($object->isSuperAdmin() && !$token->getUser()->isSuperAdmin()) {
+            if ($this->supportsAttribute($attribute) && $subject instanceof UserInterface && $token->getUser() instanceof UserInterface) {
```
Comment (jlamur) [inline Security/Authorization/Voter/UserAclVoter.php]: Please get rid of `$subject instanceof UserInterface`, this is already checked at line 42.

BTW, I'm pretty sure that this voter should deny when the token user is not an instance `UserInterface` and the `$subject` is a super admin. But this is not the purpose of this PR anyway.

Comment (jenyak) [reply Security/Authorization/Voter/UserAclVoter.php]: Could it be that `supportsClass` is overridden and support not only `UserInterface`?

Comment (jenyak) [reply Security/Authorization/Voter/UserAclVoter.php]: e.g. 
```
public function supportsClass($class)
{
    return is_subclass_of($class, 'FOS\UserBundle\Model\UserInterface') || is_subclass_of($class, 'MyClass');
}
```

**Comment by jordisala1991** [comment]
Fixed the original issue in: https://github.com/sonata-project/SonataAdminBundle/pull/5086

Closing for now. If you still think this PR is relevant, please ping me to reopen.

**Pull request closed without merging**